src/components/dev-hub/tab-context.js

```diff
@@ -0,0 +1,52 @@
+import React, { useCallback, useEffect, useReducer, useState } from 'react';
+import { getLocalValue, setLocalValue } from '../../utils/browser-storage';
+
+const defaultContextValue = {
+    activeTabs: {},
+    setActiveTab: () => {},
+};
+
+const TabContext = React.createContext(defaultContextValue);
+
+// Keeps track of the preferences for all tab sets
+const reducer = (tabSelectionPreferences, { tabset, value }) => ({
+    ...tabSelectionPreferences,
+    [tabset]: value,
+});
+
+const TabProvider = ({ children }) => {
+    const [debouncedTabUpdate, setDebouncedTabUpdate] = useState(null);
+    const [activeTabs, setActiveTab] = useReducer(
+        reducer,
+        getLocalValue('activeTabs') || defaultContextValue.activeTabs
+    );
+
+    /*
+      The LeafyGreen Tabs Implementation keeps adding key event handlers
+      to the DOM. By using a setTimeout we can avoid duplicate events being
+      run and so key events work as expected
+    */
+    const debouncedSetActiveTab = useCallback(
```

Comment:
One issue is if we have multiple sets of tabs on the page. Right now LeafyGreen adds a new event handler every time we render a `Tabs` component, meaning if we have three `Tabs` on the page then they add three separate event handlers for a key press.

This led to an issue where pressing an arrow key would not swap tabs properly. The solution for now is to debounce these on the event loop so we only get one event at a time. Should an article use tabs with different sets of tabs from each other, this could not work as anticipated (some information will be lost), but I'm willing to revisit this soon if it is an issue.

[Here](https://github.com/mongodb/leafygreen-ui/blob/master/packages/tabs/src/Tabs.tsx#L167) is the code where they add an event handler once for each tab, causing the problem.